moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryProjectTypes.js

```diff
@@ -0,0 +1,190 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Checkbox,
+  Grid,
+  Icon,
+  Input,
+  ListItemText,
+  MenuItem,
+  Select,
+  Typography,
+} from "@material-ui/core";
+
+import ProjectSummaryLabel from "./ProjectSummaryLabel";
+import { useMutation } from "@apollo/client";
+import { PROJECT_UPDATE_TYPES } from "../../../../queries/project";
+
+/**
+ * ProjectSummaryProjectTypes Component
+ * @param {Number} projectId - The id of the current project being viewed
+ * @param {Object} data - The data object from the GraphQL query
+ * @param {function} refetch - The refetch function from apollo
+ * @param {Object} classes - The shared style settings
+ * @param {function} snackbarHandle - The function to show the snackbar
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const ProjectSummaryProjectTypes = ({
+  projectId,
+  data,
+  refetch,
+  classes,
+  snackbarHandle,
+}) => {
+  /**
+   * Helper initial state lists
+   */
+  const typeList = data?.moped_types ?? [];
+  const typeDict = typeList.reduce(
+    (prev, curr) => ({
+      ...prev,
+      ...{ [curr.type_id]: curr.type_name },
+    }),
+    {}
+  );
+
+  // Original Types: array of moped_type objects
+  const originalTypes = data?.moped_project[0]?.moped_project_types ?? [];
+  // Original Types Ids: array of ids (ints)
+  const originalTypesIds = originalTypes.map(t => t?.moped_type?.type_id);
+
+  /**
+   * Edit Mode and selected Types states
+   */
+  const [editMode, setEditMode] = useState(false);
+  const [selectedTypes, setSelectedTypes] = useState(originalTypesIds);
+
+  // The mutation and mutation function
+  const [updateProjectTypes] = useMutation(PROJECT_UPDATE_TYPES);
+
+  /**
+   * Handles whenever there is a click in any of the menu items
+   * @param {Object} event - The event object
+   */
+  const handleChange = event => {
+    setSelectedTypes(event.target.value);
+  };
+
+  /**
+   * Resets the types list back to the original state, closes edit mode
+   */
+  const handleProjectTypesClose = () => {
+    setSelectedTypes(originalTypesIds);
+    setEditMode(false);
+  };
+
+  /**
+   * Saves the new project types
+   */
+  const handleProjectTypesSave = () => {
+    // Original type Ids
+    const oldTypesList = originalTypesIds;
+    // Get selected types ids
+    const newTypesList = selectedTypes;
+    // Retrieves the ids of oldTypesList that are not present in newTypesList
+    const typeIdsToDelete = oldTypesList.filter(t => !newTypesList.includes(t));
+    // Retrieves the ids of newTypesList that are not present in oldTypesList
+    const typeIdsToInsert = newTypesList.filter(t => !oldTypesList.includes(t));
```

Comment:
Another throwback for me, maybe there is a way to improve the O(n*n) performance, but the list is so small its kind of irrelevant. 👍  Feel free to challenge these as you code.